### ISSUE TITLE
How do you get Metabase to use getdate() instead of getutcdate()?

### ISSUE THREAD

**Issue by jkowtko** [metabase/metabase#8326]
Our server is running on Eastern Time, no overrides in SQL Server.  Metabase timezone is also set to Eastern Time.

In my "Question" the date comparison filters are generating SQL with getutcdate() in it ... they need to use getdate() in order for the query observe the correct day threshold.

What is the secret to stop Metabase from assuming our database is storing dates in UTC?

Thanks.  John

===
Metabase version v0.29.3 Built on 2018-05-12
Microsoft SQL Server 2014 (SP2-GDR) (KB4019093) - 12.0.5207.0 (X64) 
	Jul  3 2017 02:25:44 
	Copyright (c) Microsoft Corporation
	Express Edition (64-bit) on Windows NT 6.1 <X64> (Build 7601: Service Pack 1) (Hypervisor)
Windows Server 2008 R2 Standard
===

**Comment by jornh** [comment]
Pong John,

Just another Metabase user here. Not currently using SQL Server myself. It appears the offending line could be this one with `getutcdate()`:

https://github.com/metabase/metabase/blob/b6d0bfcd7bfa21370bfc86954b8e18f65aaa5ad7/src/metabase/driver/sqlserver.clj#L140

Now the question becomes if there could be any negative consequences of changing this line? And how to test this?

**Comment by salsakran** [comment]
@jkowtko Can you share a specific query example? 

Also, is there a specific user facing bug in how we the filter are applied that you are noticing? I.e. are we producing wrong numbers when filtering or is this more of an abstract concern?

**Comment by jkowtko** [comment]
Specifically my query filters on "date = today" ... this returns no records after 8pm ET because getutcdate() has just passed midnight and the query is now looking for data that is dated tomorrow.   We don't generate tomorrow's data until tomorrow around 1-2am.
Thanks.  John
John S Kowtko 
650-533-1222 

    On Wednesday, October 31, 2018, 12:08:17 PM PDT, Sameer Al-Sakran <notifications@github.com> wrote:  
 
 
@jkowtko Can you share a specific query example?

Also, is there a specific user facing bug in how we the filter are applied that you are noticing? I.e. are we producing wrong numbers when filtering or is this more of an abstract concern?

**Comment by camsaul** [comment]
If we switch this to `getdate` can you guys think of anything that would break? If not happy to make the change

**Comment by jkowtko** [comment]
If you're asking me, then in my limited experience with Metabase I know of nothing that would want us to have our Metabase instance operating in anything other than the local server timezone.

Quick poking around and I found https://github.com/metabase/metabase/pull/5729 ... at first glance the was an issue with datetime fields without timezone information being handled incorrectly by Metabase.  Is it possible the the "fix" was that Metabase will assume UTC timezone is in effect for any datetime field that doesn't have timezone information associated with it?  That obviously is not a correct assumption.

I also have not seen anything in Metabase that visibly allows you to distinguish between UTC and non-UTC datetime reference ... so IMHO there should not be any getutcdate() function calls being generated automatically by Metabase.

Thanks.  John

**Comment by RMMSecurity** [comment]
I'm testing this from master now. I'll see if I notice anything breaking.

**Comment by notrom** [comment]
I'm in "Pacific/Auckland" timezone and see the same issue. Can cause significant confusion when using relative date filters like "Today" or "Yesterday".

I took a quick poke around the Metabase source code for the Oracle driver and see that it uses the "NOW" function which I think is equivalent to SQL Server's getdate() not getutcdate()

https://github.com/metabase/metabase/blob/14dd7d8b3ebf482e2fe2f004fdec038c29d13808/modules/drivers/oracle/src/metabase/driver/oracle.clj#L130

Similarly for the Redshift driver, I think Redshift's getdate() returns the local timezone not UTC

https://github.com/metabase/metabase/blob/c1297b494f25c4859c128850c06565a71faf19ba/modules/drivers/redshift/src/metabase/driver/redshift.clj#L85-L86

Making the proposed change could make the SQL Server driver more consistent with the other drivers?

**Comment by camsaul** [comment]
Implemented by #9248

**Comment by flamber** [comment]
@camsaul 
I think you missed the last `getutcdate` on line [221](https://github.com/metabase/metabase/blob/master/modules/drivers/sqlserver/src/metabase/driver/sqlserver.clj#L221) because on 0.32.1 I just did a QB filtering on date-column=Today and didn't get any results, then I checked the SQL, which was:
```
WHERE CAST(CAST("dbo"."Facturas"."fecha" AS date) AS datetime) = CAST(CAST(getutcdate() AS date) AS datetime) 
```
I then tried to compile a driver with `getutcdate` changed to `getdate` on line 221 and now I'm seeing results and the query is:
```
WHERE CAST(CAST("dbo"."Facturas"."fecha" AS date) AS datetime) = CAST(CAST(getdate() AS date) AS datetime) 
```

**Comment by camsaul** [comment]
Fixed by #9706
